src/ObjWriting/XModel/Export/XModelBinWriter.cpp

```diff
@@ -0,0 +1,474 @@
+        BONE_MATRIX_Y = 0xCCDC,
+        BONE_MATRIX_Z = 0xFCBF,
+
+        VERT16_COUNT = 0x950D,
+        VERT32_COUNT = 0x2AEC,
+        VERT16 = 0x8F03,
+        VERT32 = 0xB097,
+        VERT_WEIGHT_COUNT = 0xEA46,
+        VERT_WEIGHT = 0xF1AB,
+
+        FACE_COUNT = 0xBE92,
+        TRIANGLE16 = 0x6711,
+        TRIANGLE32 = 0x562F,
+
+        NORMAL = 0x89EC,
+        COLOR = 0x6DD8,
+        UV = 0x1AD4,
+
+        OBJECT_COUNT = 0x62AF,
+        OBJECT = 0x87D4,
+
+        MATERIAL_COUNT = 0xA1B2,
+        MATERIAL = 0xA700,
+        MATERIAL_TRANSPARENCY = 0x6DAB,
+        MATERIAL_AMBIENT_COLOR = 0x37FF,
+        MATERIAL_INCANDESCENCE = 0x4265,
+        MATERIAL_COEFFS = 0xC835,
+        MATERIAL_GLOW = 0xFE0C,
+        MATERIAL_REFRACTIVE = 0x7E24,
+        MATERIAL_SPECULAR_COLOR = 0x317C,
+        MATERIAL_REFLECTIVE_COLOR = 0xE593,
+        MATERIAL_REFLECTIVE = 0x7D76,
+        MATERIAL_BLINN = 0x83C7,
+        MATERIAL_PHONG = 0x5CD2
+    };
+
+protected:
+    void PrepareVertexMerger(const XModelCommon& xmodel)
+    {
+        m_vertex_merger = VertexMerger(xmodel.m_vertices.size());
+
+        auto vertexOffset = 0u;
+        for (const auto& vertex : xmodel.m_vertices)
+        {
+            XModelVertexBoneWeights weights{0, 0};
+
+            if (vertexOffset < xmodel.m_vertex_bone_weights.size())
+                weights = xmodel.m_vertex_bone_weights[vertexOffset];
+
+            m_vertex_merger.Add(VertexMergerPos{vertex.coordinates[0],
+                                                vertex.coordinates[1],
+                                                vertex.coordinates[2],
+                                                xmodel.m_bone_weight_data.weights.empty() ? nullptr : &xmodel.m_bone_weight_data.weights[weights.weightOffset],
+                                                weights.weightCount});
+
+            vertexOffset++;
+        }
+    }
+
+    void WriteAlignedString(const std::string& string)
+    {
+        const auto paddingSize = ((string.size() + 1 + 0x3) & 0xFFFFFFFFFFFFFC) - (string.size() + 1);
+        const auto padding = std::make_unique<uint8_t[]>(static_cast<size_t>(paddingSize));
+
+        m_writer.WriteNullTerminatedString(string);
+        m_writer.Write(reinterpret_cast<uint8_t*>(padding.get()), static_cast<uint32_t>(paddingSize));
+    }
+
+    void WriteComment(const std::string& comment)
+    {
+        m_writer.Write(XModelBinHash::COMMENT);
+        WriteAlignedString(comment);
+    }
+
+    void WriteInt16(const int16_t hash, const int16_t value)
+    {
+        m_writer.Write(hash);
+        m_writer.Write(value);
+    }
+
+    void WriteUInt16(const int16_t hash, const uint16_t value)
+    {
+        m_writer.Write(hash);
+        m_writer.Write(value);
+    }
+
+    int16_t ClampFloatToShort(const float value) const
+    {
+        return std::clamp(static_cast<int16_t>(32767 * std::clamp(value, -1.0f, 1.0f)), static_cast<int16_t>(-32767), static_cast<int16_t>(32767));
```

Comment:
Can you maybe use `std::numeric_limits<int16_t>::max()` and `::min()` respectively?